tests/templates/test_subroutines/test_multiplier.py

```diff
@@ -172,9 +160,17 @@ def test_operation_and_wires_error(
     def test_decomposition(self):
         """Test that compute_decomposition and decomposition work as expected."""
         k, x_wires, mod, work_wires = 4, [0, 1, 2], 7, [3, 4, 5, 6, 7]
-        multiplier_decomposition = qml.Multiplier(
-            k, x_wires, mod, work_wires
-        ).compute_decomposition(k, x_wires, mod, work_wires)
+        _multiplier_decomposition = (
+            qml.Multiplier(k, x_wires, mod, work_wires)
+            .compute_decomposition(k, x_wires, mod, work_wires)[0]
+            .decomposition()
+        )
+        multiplier_decomposition = [
+            *_multiplier_decomposition[0].decomposition(),
+            *_multiplier_decomposition[1].decomposition(),
+            *_multiplier_decomposition[2].decomposition(),
+        ]
```

Comment:
Could we somehow use an expansion strategy with `max_expansion=3` or so? Could be a bit more compact/clear what is going on.

Reply:
Neat! Applied the suggestion.